packages/dev/docs/src/client.js

```diff
@@ -86,3 +86,23 @@ ReactDOM.render(<>
   <Hamburger />
   <ThemeSwitcher />
 </>, document.getElementById('header'));
+
+document.addEventListener('mousedown', (e) => {
+  // Prevent focusing on links to other pages with the mouse to avoid flash of focus ring during navigation.
+  let link = e.target.closest('a');
+  if (link && (link.host !== location.host || link.pathname !== location.pathname)) {
+    e.preventDefault();
+  }
+
+  // Add mouse focus class to summary elements on mouse down to prevent native browser focus from showing.
+  if (e.target.tagName === 'SUMMARY') {
+    e.target.classList.add(docsStyle.mouseFocus);
+  }
+});
+
+// Remove mouse focus class on blur of a summary element.
+document.addEventListener('blur', (e) => {
+  if (e.target.tagName === 'SUMMARY') {
+    e.target.classList.remove(docsStyle.mouseFocus);
+  }
+}, true);
```

Comment:
Curious why we would do this to mimic focus-ring behavior, rather than eat our own dog food?

Reply:
well... it's pretty hard as you saw to actually do this without re-rendering everything on the client that we already rendered on the server. Seems like a lot of extra JS just for the focus ring.